mmdet/utils/deployment/openvino_backend.py

```diff
@@ -242,3 +246,105 @@ def __call__(self, inputs, **kwargs):
+        self.hidden_shape = [v.shape for k, v in self.text_decoder.net.inputs.items() if k == 'prev_hidden'][0]
+        self.vocab_size = [v.shape[-1] for k, v in self.text_decoder.net.outputs.items() if k == 'output'][0]
+        self.alphabet='  ' + string.ascii_lowercase + string.digits
+
+
+    def configure_outputs(self, required):
+        extra_outputs = ['boxes', 'labels', 'masks', 'text_features']
+
+        for output in extra_outputs:
+            if output not in self.orig_ir_mapping and output in self.net.outputs:
+                self.orig_ir_mapping[output] = output
+            print(self.orig_ir_mapping[output], output)
+
+        self.try_add_extra_outputs(extra_outputs)
+        outputs = []
+
+        self.check_required(self.orig_ir_mapping.keys(), ['boxes', 'labels', 'text_features'])
+        self.with_detection_output = False
+        outputs.extend(['boxes', 'labels', 'text_features'])
+
+        try:
+            self.check_required(self.orig_ir_mapping.keys(), ['masks'])
+            self.with_mask = True
+            outputs.append('masks')
+        except ValueError:
+            self.with_mask = False
+
+        self.set_outputs(outputs)
+
+    def __call__(self, inputs, **kwargs):
+        inputs = self.unify_inputs(inputs)
+        data_h, data_w = inputs['image'].shape[-2:]
+        inputs['image'] = np.pad(inputs['image'],
+                                 ((0, 0), (0, 0), (0, self.h - data_h), (0, self.w - data_w)),
+                                 mode='constant')
+
+        output = super().__call__(inputs)
+
+        valid_detections_mask = output['boxes'][:,-1] >= 0.5
+        output['labels'] = output['labels'][valid_detections_mask]
+        output['boxes'] = output['boxes'][valid_detections_mask]
+        output['text_features'] = output['text_features'][valid_detections_mask]
+
+        if 'masks' in output:
+            output['masks'] = output['masks'][valid_detections_mask]
+
+        texts = []
+        for feature in output['text_features']:
+            feature = np.expand_dims(feature, 0)
+            feature = self.text_encoder({'input': feature})['output']
+            feature = np.reshape(feature, (feature.shape[0], feature.shape[1], -1))
+            feature = np.transpose(feature, (0, 2, 1))
+
+            hidden = np.zeros(self.hidden_shape)
+            prev_symbol = np.zeros((1,))
+
+            eos = 1
+            max_seq_len = 28
+            per_feature_outputs = np.zeros([max_seq_len, self.vocab_size])
+
+            decoded = ''
+            confidence = 1
+
+            for i in range(max_seq_len):
+                out = self.text_decoder({
+                    'prev_symbol': prev_symbol,
+                    'prev_hidden': hidden,
+                    'encoder_outputs': feature
+                })
+                per_feature_outputs[i] = out['output']
+                softmaxed = softmax(out['output'], axis=1)
```

Comment:
What prevents this softmax from being a part of the model?

Reply:
Nothing, I'll make it for next releases. Previously I didn't use confidence so softmax was not needed.